File: src/balancedround.py
t = int(input())
for _ in range(t):
    n, k = map(int, input().split())
    arr = list(map(int, input().split()))
    arr.sort()
    ans = 0; i = 0
    while i < n:
        j = i + 1
        while j < n:
            if arr[j] - arr[j - 1] > k: 
                break
            j += 1
        tt = j - i; ans = max(ans, tt); i = j
    print(n - ans)

File: src/balancedround.cpp
#include<bits/stdc++.h>
using namespace std;

int main(){
    int t; cin >> t;
    while(t--){
        int n, k; cin >> n >> k;
        int arr[n];
        for(int i = 0; i < n; i++) cin >> arr[i];
        sort(arr, arr + n);
        int ans = 0;
        int i = 0;
        while(i < n){
            int j = i + 1;
            while(j < n){
                if(arr[j] - arr[j - 1] > k) break;
                j++;
            }
            int tt = j - i;
            ans = max(ans, tt); i = j;
        }
        cout << n - ans << endl;
    }
    return 0;
}